src/lib/support/Pool.h

```diff
@@ -160,4 +167,107 @@ class BitMapObjectPool : public StaticAllocatorBitmap
     } mData;
 };
 
+#if CHIP_SYSTEM_CONFIG_POOL_USE_HEAP
+
+/**
```

Comment:
This looks a lot like the implementation in SystemObject....did that go away?

Reply:
Not yet; I will do that in a separate PR. (The `System::Object` one differs primarily in using `Object` for the collection of objects.)

Reply:
OH! yeah, that's better. OK, in that case, I think the swaps I requested (s/mutex/SystemMutex et al.) are out of scope for this then.